### ISSUE TITLE
pdf printing crash in 10.1.5: "Cannot read property 'setDockSide' of undefined"

### ISSUE THREAD

**Issue by SUH11** [electron/electron#26587]
### Preflight Checklist

* [x] I have read the [Contributing Guidelines](https://github.com/electron/electron/blob/master/CONTRIBUTING.md) for this project.
* [x] I agree to follow the [Code of Conduct](https://github.com/electron/electron/blob/master/CODE_OF_CONDUCT.md) that this project adheres to.
* [x] I have searched the issue tracker for an issue that matches the one I want to file, without success.

### Issue Details

* **Electron Version:**
  * 9.3.4 10.1.5 10.1.3 10.1.5  11.0.1 11.0.0-beta.23
* **Operating System:**
  * window10 and mac 10.14.6 
* **Last Known Working Electron version:**
  * 8.5.4

### Expected Behavior
save as pdf or preview it can work

### Actual Behavior
crash

### To Reproduce
```javascript
// main.js
const { app, BrowserWindow } = require('electron')

function createWindow() {
    const mainWindow = new BrowserWindow({
        width: 1200, 
        height: 700,
        webPreferences: {
            webviewTag: true,
            nativeWindowOpen: true,
            enableRemoteModule: true
        }
    })
    mainWindow.loadURL(`file://${__dirname}/index.html`)
    mainWindow.webContents.openDevTools()
    console.log('i am ready')
    setTimeout(() =>{
        // mainWindow.webContents.print()
    }, 5000)
}
app.on('ready', createWindow)

```

```html
<!DOCTYPE html>
<html>
  <head>
    <meta charset="UTF-8">
    <title>Hello World!</title>
  </head>
  <body>
    <h1 onclick="myPrint()">print me</h1>
    <script>
      function myPrint(){
        let iframe = document.createElement('iframe')
        iframe.setAttribute("style", "width:100%; height:100%; visibility:visible; overflow:hidden;")
        iframe.setAttribute('src', './1.pdf') //  just PDF crash...
        iframe.type = "application/pdf"
        const oBody = document.getElementsByTagName('body')[0]
        oBody.appendChild(iframe)
        iframe.contentWindow.print()
      }
    </script>
  </body>
</html>
```

```sh
$ git clone git@github.com:SUH11/electron-hello-world.git
$ npm install
$ npm start || electron .
```

### Screenshots
![image](https://user-images.githubusercontent.com/16281522/99614412-68df0d80-2a54-11eb-9be6-b2753ec26c6d.png)

**Comment by ckerr** [comment]
Confirmed with @SUH11's testcase repo. Here's the crash I get on Electron 11 + Ubuntu 20.04:

```
i am ready
(node:1025145) electron: The default of contextIsolation is deprecated and will be changing from false to true in a future release of Electron.  See https://github.com/electron/electron/issues/23506 for more information
[1025145:1119/104942.898409:ERROR:CONSOLE(1)] "Uncaught TypeError: Cannot read property 'setDockSide' of undefined", source: devtools://devtools/bundled/devtools_app.html?remoteBase=https://chrome-devtools-frontend.appspot.com/serve_file/@0cb1632d85c05d3b7fdd4441146bd58126a253a/&can_dock=true&toolbarColor=rgba(223,223,223,1)&textColor=rgba(0,0,0,1)&experiments=true (1)
[1025187:1119/105004.675144:FATAL:render_frame_impl.cc(2432)] Check failed: !base::RunLoop::IsNestedOnCurrentThread(). 
#0 0x5630d16544e9 base::debug::CollectStackTrace()
#1 0x5630d15a2443 base::debug::StackTrace::StackTrace()
#2 0x5630d15b884f logging::LogMessage::~LogMessage()
#3 0x5630d15b90ce logging::LogMessage::~LogMessage()
#4 0x5630d5064aa0 content::RenderFrameImpl::OnUnload()
#5 0x5630d5064945 IPC::MessageT<>::Dispatch<>()
#6 0x5630d506468e content::RenderFrameImpl::OnMessageReceived()
#7 0x5630d19ae2c1 IPC::ChannelProxy::Context::OnDispatchMessage()
#8 0x5630d0670bd1 base::internal::Invoker<>::RunOnce()
#9 0x5630d15fdb4b base::TaskAnnotator::RunTask()
#10 0x5630d1616d1c base::sequence_manager::internal::ThreadControllerWithMessagePumpImpl::DoWorkImpl()
#11 0x5630d1616948 base::sequence_manager::internal::ThreadControllerWithMessagePumpImpl::DoWork()
#12 0x5630d15bd5d4 base::MessagePumpDefault::Run()
#13 0x5630d16176d4 base::sequence_manager::internal::ThreadControllerWithMessagePumpImpl::Run()
#14 0x5630d15e1985 base::RunLoop::Run()
#15 0x5630d2f87e4f IPC::SyncChannel::WaitForReplyWithNestedMessageLoop()
#16 0x5630d2f87d45 IPC::SyncChannel::WaitForReply()
#17 0x5630d2f875fa IPC::SyncChannel::Send()
#18 0x5630d302536f content::ChildThreadImpl::Send()
#19 0x5630d514d405 content::RenderThreadImpl::Send()
#20 0x5630d5bad388 printing::PrintRenderFrameHelper::GetPrintSettingsFromUser()
#21 0x5630d5ba6001 printing::PrintRenderFrameHelper::Print()
#22 0x5630d5ba41b8 printing::PrintRenderFrameHelper::ScriptedPrint()
#23 0x5630d50640ad content::RenderFrameImpl::ScriptedPrint()
#24 0x5630d42387e1 blink::ChromeClient::Print()
#25 0x5630d3985db3 blink::LocalDOMWindow::print()
#26 0x5630d39b5656 blink::LocalFrame::FinishedLoading()
#27 0x5630d4199f55 blink::FrameLoader::DidFinishNavigation()
#28 0x5630d31dd283 blink::TimerBase::RunInternal()
#29 0x5630cdad45db base::internal::Invoker<>::Run()
#30 0x5630d045bbf1 WTF::ThreadCheckingCallbackWrapper<>::Run()
#31 0x5630d15fdb4b base::TaskAnnotator::RunTask()
#32 0x5630d1616d1c base::sequence_manager::internal::ThreadControllerWithMessagePumpImpl::DoWorkImpl()
#33 0x5630d1616948 base::sequence_manager::internal::ThreadControllerWithMessagePumpImpl::DoWork()
#34 0x5630d15bd5d4 base::MessagePumpDefault::Run()
#35 0x5630d16175ac base::sequence_manager::internal::ThreadControllerWithMessagePumpImpl::Run()
#36 0x5630d15e1985 base::RunLoop::Run()
#37 0x5630d59e9c92 content::RendererMain()
#38 0x5630d05f46ee content::ContentMainRunnerImpl::Run()
#39 0x5630cebdfebc content::RunContentProcess()
#40 0x5630cebdff9c content::ContentMain()
#41 0x5630cdaa4b6e main
#42 0x7f16dc7f10b3 __libc_start_main
#43 0x5630cda8d5aa _start
Task trace:
#0 0x5630d19ae185 IPC::ChannelProxy::Context::OnMessageReceivedNoFilter()
IPC message handler context: 0xD60BEA4C

Received signal 6
#0 0x5630d16544e9 base::debug::CollectStackTrace()
#1 0x5630d15a2443 base::debug::StackTrace::StackTrace()
#2 0x5630d165403b base::debug::(anonymous namespace)::StackDumpSignalHandler()
#3 0x7f16ddfc43c0 (/lib/x86_64-linux-gnu/libpthread-2.31.so+0x153bf)
#4 0x7f16dc81018b gsignal
#5 0x7f16dc7ef859 abort
#6 0x5630d1652e15 base::debug::BreakDebugger()
#7 0x5630d15b8ca0 logging::LogMessage::~LogMessage()
#8 0x5630d15b90ce logging::LogMessage::~LogMessage()
#9 0x5630d5064aa0 content::RenderFrameImpl::OnUnload()
#10 0x5630d5064945 IPC::MessageT<>::Dispatch<>()
#11 0x5630d506468e content::RenderFrameImpl::OnMessageReceived()
#12 0x5630d19ae2c1 IPC::ChannelProxy::Context::OnDispatchMessage()
#13 0x5630d0670bd1 base::internal::Invoker<>::RunOnce()
#14 0x5630d15fdb4b base::TaskAnnotator::RunTask()
#15 0x5630d1616d1c base::sequence_manager::internal::ThreadControllerWithMessagePumpImpl::DoWorkImpl()
#16 0x5630d1616948 base::sequence_manager::internal::ThreadControllerWithMessagePumpImpl::DoWork()
#17 0x5630d15bd5d4 base::MessagePumpDefault::Run()
#18 0x5630d16176d4 base::sequence_manager::internal::ThreadControllerWithMessagePumpImpl::Run()
#19 0x5630d15e1985 base::RunLoop::Run()
#20 0x5630d2f87e4f IPC::SyncChannel::WaitForReplyWithNestedMessageLoop()
#21 0x5630d2f87d45 IPC::SyncChannel::WaitForReply()
#22 0x5630d2f875fa IPC::SyncChannel::Send()
#23 0x5630d302536f content::ChildThreadImpl::Send()
#24 0x5630d514d405 content::RenderThreadImpl::Send()
#25 0x5630d5bad388 printing::PrintRenderFrameHelper::GetPrintSettingsFromUser()
#26 0x5630d5ba6001 printing::PrintRenderFrameHelper::Print()
#27 0x5630d5ba41b8 printing::PrintRenderFrameHelper::ScriptedPrint()
#28 0x5630d50640ad content::RenderFrameImpl::ScriptedPrint()
#29 0x5630d42387e1 blink::ChromeClient::Print()
#30 0x5630d3985db3 blink::LocalDOMWindow::print()
#31 0x5630d39b5656 blink::LocalFrame::FinishedLoading()
#32 0x5630d4199f55 blink::FrameLoader::DidFinishNavigation()
#33 0x5630d31dd283 blink::TimerBase::RunInternal()
#34 0x5630cdad45db base::internal::Invoker<>::Run()
#35 0x5630d045bbf1 WTF::ThreadCheckingCallbackWrapper<>::Run()
#36 0x5630d15fdb4b base::TaskAnnotator::RunTask()
#37 0x5630d1616d1c base::sequence_manager::internal::ThreadControllerWithMessagePumpImpl::DoWorkImpl()
#38 0x5630d1616948 base::sequence_manager::internal::ThreadControllerWithMessagePumpImpl::DoWork()
#39 0x5630d15bd5d4 base::MessagePumpDefault::Run()
#40 0x5630d16175ac base::sequence_manager::internal::ThreadControllerWithMessagePumpImpl::Run()
#41 0x5630d15e1985 base::RunLoop::Run()
#42 0x5630d59e9c92 content::RendererMain()
#43 0x5630d05f46ee content::ContentMainRunnerImpl::Run()
#44 0x5630cebdfebc content::RunContentProcess()
#45 0x5630cebdff9c content::ContentMain()
#46 0x5630cdaa4b6e main
#47 0x7f16dc7f10b3 __libc_start_main
#48 0x5630cda8d5aa _start
  r8: 0000000000000000  r9: 00007fffd063a900 r10: 0000000000000008 r11: 0000000000000246
 r12: 00007fffd063ab60 r13: aaaaaaaaaaaaaaaa r14: 0000192611bffb40 r15: 00007fffd063ab60
  di: 0000000000000002  si: 00007fffd063a900  bp: 00007fffd063ab50  bx: 00007f16db495d80
  dx: 0000000000000000  ax: 0000000000000000  cx: 00007f16dc81018b  sp: 00007fffd063a900
  ip: 00007f16dc81018b efl: 0000000000000246 cgf: 002b000000000033 erf: 0000000000000000
 trp: 0000000000000000 msk: 0000000000000000 cr2: 0000000000000000
[end of stack trace]
Calling _exit(1). Core file will not be generated.
```

**Comment by FrankWangZY** [comment]
It happened in 9.3.4 either, please help to fix it ,thanks!

**Comment by pushkin-** [comment]
for the record, I don't experience this crash on Windows 10 with Electron 11.1.0. (I'm guessing you're talking about the iframe print, not the commented-out webContents.print.

**Comment by SUH11** [comment]
> for the record, I don't experience this crash on Windows 10 with Electron 11.1.0. (I'm guessing you're talking about the iframe print, not the commented-out webContents.print.

yes, not the commented-out webContents.print.  just print pdf, it crash in Windows with Electron 11.1.0.(or some other versions i listed.)

**Comment by weifeiyue** [comment]
When can it be fixed, I was urged to death by the user😭😭😭😭😭😭

**Comment by weifeiyue** [comment]
> > for the record, I don't experience this crash on Windows 10 with Electron 11.1.0. (I'm guessing you're talking about the iframe print, not the commented-out webContents.print.
> 
> yes, not the commented-out webContents.print. just print pdf, it crash in Windows with Electron 11.1.0.(or some other versions i listed.)

When can it be fixed, I was urged to death by the user

**Comment by weifeiyue** [comment]
@ckerr can u help me to fix it ? please~ 😭😭😭

**Comment by codebytere** [comment]
Closed by https://github.com/electron/electron/pull/26474

**Comment by SUH11** [comment]
> Closed by #26474

Hi, I change the electron version to 10.0.0-beta.25 (or 10.4.5), printing PDF will still crash. But I see a [commit ID](https://github.com/electron/electron/pull/27691) to fix this issue. So, this issue has been fixed? Help please...
